.github/workflows/test_code.yml

```diff
@@ -12,13 +12,13 @@ jobs:
     steps:
     - uses: actions/checkout@v2
     - name: Cache pip
-      uses: actions/cache@v2
+      uses: actions/cache@v4
```

Comment:
Is this an accidental change - If this is needed for our pipelines to succeed, maybe create a seperate PR?

Reply:
Apparently I didn't read the full PR description, disregard my comment